### PR TITLE
Animal sample table options

### PR DESCRIPTION
**Pull request by fkang-pu**
## Summary Change Description

- modified base.html to use CDN links for Bootstrap css and JS, upgraded to version 5.0.2 from 5.0.1. Included css and JavaScript required by Bootstrap Table plugin.

- Added table options (search, filter columns, full screen, show/hide/columns, export table to file) to Animal and Sample List.

## Affected Issue Numbers

- partly addresses #216 
- partly addresses  #214
- partly addresses  #215

## Code Review Notes

Only selected a few Bootstrap Table options and applied to animal and sample lists. Need careful review before making similar changed for other object lists.

## Checklist

- [ ] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- [x] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
- [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)

### CODE REVIEW

File: DataRepo/templates/DataRepo/animal_list.html [hunk 1]
```diff
@@ -9,41 +9,58 @@ <h4>List of Animals</h4>
...
+                <th data-field="Tracer" data-filter-control="input" data-sortable="true">Tracer(Infusate)</th>
+                <th data-field="Labeled Atom" data-filter-control="input" data-sortable="true">Labeled Atom</th>
+                <th data-field="Labeled Count" data-sortable="true">Labeled Count</th>
+                <th>Infusion Rate</th>
+                <th>Infusion Concentration </th>
+                <th data-field="Genotype" data-filter-control="select" data-sortable="true">Genotype</th>
+                <th>Body Weight</th>
+                <th>Age</th>
+                <th>Sex</th>
+                <th>Diet</th>
```
Comment (lparsons) [inline DataRepo/templates/DataRepo/animal_list.html:38]: NB: I don't see a reason not to make these sortable. I would also consider filters on them as well, with Sex having a select filter, the other text.

**Review by lparsons — APPROVED** [review]
This looks very polished and the functionality is very useful and applicable to many tables in the application. 

I put a few thoughts/suggestions for specific columns inline. Another small nit is that the search box can sometimes end up to the right of the table. One solution would be to make the table take up the full width of the container. I *think* that could be done by [putting the table in a container `div` with the class `table-responsive`](https://stackoverflow.com/a/28339204/63093), but I'm not certain that holds true for BS 5.

image

These are all small suggestions, however. I approve of merging this with some or no changes and applying it to other tables in future PRs.

**Review by jcmatese — APPROVED** [review]
Agree with Lance the JSON is probably not needed as a downloadable format at this time.  Other than that, we could probably enable widgets when asked for, given usage and user experience.

**Comment by fkang-pu** [comment]
Pushed a commit this morning. Moved search box to left side of the table, made every column sortable, and removed json type for file download.

File: DataRepo/templates/DataRepo/animal_list.html [hunk 2]
```diff
@@ -9,41 +9,58 @@ <h4>List of Animals</h4>
...
+            data-show-columns="true"
+            data-show-columns-toggle-all="true"
+            data-show-fullscreen="true"
+            data-show-export="true"
+            data-export-types="['csv', 'txt', 'excel', 'json']">
+            <thead>
+                <tr>
+                <th data-field="Name" data-filter-control="input" data-sortable="true">Name</th>
+                <th data-field="Tracer" data-filter-control="input" data-sortable="true">Tracer(Infusate)</th>
+                <th data-field="Labeled Atom" data-filter-control="input" data-sortable="true">Labeled Atom</th>
```
[the remark this answers is not in the document]
Comment (lparsons) [reply DataRepo/templates/DataRepo/animal_list.html:31]: It seems like the problem was actually having spaces in the `data-field` name. I sent you a PR that fixes that uses `select` in a few places that made sense to me.

**Comment by fkang-pu** [comment]
@lparsons Rob hasn't reviewed this PR yet, maybe we merge your #220 into my romote branch first, I then ask re-review. You do the merge or you want me to do it?

**Pull request merged**